### PR TITLE
TST/REF: Add more pytest idiom to tests/tslib

### PR DESCRIPTION
**Pull request by gfyoung**
I was planning to correct individual files, but then I began to realize that the fixes were involving more and more files (was shifting tests around) in the directory, to the point that I just modified them all. 🙂

### CODE REVIEW

File: pandas/tests/tslibs/test_conversion.py [hunk 1]
```diff
@@ -11,61 +11,62 @@
...
-        if copy:
-            assert result.base is None
-        else:
-            assert result.base is arr
+    expected = np.vectorize(f)(utc_didx.asi8)
+
+    tm.assert_numpy_array_equal(result, expected)
+
+
+@pytest.fixture(params=["UTC", "Asia/Tokyo",
```
Comment (jreback) [inline pandas/tests/tslibs/test_conversion.py]: can you use tz_aware_fixture instead

File: pandas/tests/tslibs/test_liboffsets.py [hunk 2]
```diff
@@ -12,161 +12,163 @@
...
+            "end": 1,
+            "business_end": 1
+        },
+        -1: {
+            "start": 0
+        }
+    })
+])
+@pytest.mark.parametrize("n", [2, -1])
+@pytest.mark.parametrize("day_opt", ["start", "end",
```
Comment (jreback) [inline pandas/tests/tslibs/test_liboffsets.py]: maybe a day_opt fixture (in this file)

**Review by jreback — CHANGES_REQUESTED** [review]
looks good. some comments.

**Comment by jbrockmendel** [comment]
Is this just getting rid of a level of indentation by removing the classes?  I get that they don't serve much functional purpose, but do find them helpful for organization.  Is there a preferred idiom for organization?

**Comment by gfyoung** [comment]
@jbrockmendel : In general, small, modular, functional test cases would be preferred I think (both for consistency with pytest idiom and readability, especially the short and modular part).   I agree that test classes can be useful, especially when we are sharing test cases amongst multiple files or the test file is massive and has several super distinct collections of tests.

That being said, for this case, I did not really find that to be the case.  Note that this PR attempts to reorganize the tests a little as well as add a lot more parameterization (in interests of the short and modular).

I don’t think we will have a unified style for tests.  We might prefer functional, but I wouldn’t force it everywhere if the (semantic) organization does not dictate it to be so.

**Comment by gfyoung** [comment]
@jreback : Addressed all comments, and everything is green.  PTAL.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @gfyoung

**Pull request merged**